grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/Atom.java

```diff
@@ -212,10 +227,12 @@ public int baseResolutionPriority(){
     }
 
     /**
+     * @param type the class of {@link Predicate} to return
+     * @param <T> the type of neighbour {@link Atomic} to return
      * @return neighbours of this atoms, i.e. atoms connected to this atom via shared variable
      */
-    public Stream<Atom> getNeighbours(){
-        return getParentQuery().getAtoms(Atom.class)
+    public <T extends Atomic> Stream<T> getNeighbours(Class<T> type){
```

Comment:
If you wanted you could make this take in an optional array:

```
getNeighbours(Class<T> . . . types)
```

could make the method more versatile. Just a suggestion though.